packages/validator/src/services/attestationDuties.ts

```diff
@@ -63,12 +63,32 @@ export class AttestationDutiesService {
 
     if (metrics) {
       metrics.attesterDutiesCount.addCollect(() => {
+        const currentSlot = this.clock.getCurrentSlot();
         let duties = 0;
-        for (const attDutiesAtEpoch of this.dutiesByIndexByEpoch.values()) {
+        let nextDutySlot = null;
+        for (const [epoch, attDutiesAtEpoch] of this.dutiesByIndexByEpoch) {
           duties += attDutiesAtEpoch.dutiesByIndex.size;
+
+          // Historical epochs can be skipped when determining next duty slot
+          if (epoch < this.clock.currentEpoch) continue;
```

Comment:
@nflaig I mean once we found `nextDutySlot` and since epochs are sorted, we can do
```suggestion
          if (epoch < this.clock.currentEpoch || nextDutySlot !== null) continue;
```

to skip the for loop below

Reply:
oh right, makes sense. Will fix that